src/ZiraLink.Api.Application/Tools/HttpTools.cs

```diff
@@ -48,5 +58,52 @@ public async Task<bool> PingDomain(string domainUrl)
 
             return true;
         }
+
+
+        public async Task<HttpClient> InitializeHttpClientAsync(CancellationToken cancellationToken)
+        {
+            var httpClient = _httpClientFactory.CreateClient(NamedHttpClients.Default);
+            var disco = await httpClient.GetDiscoveryDocumentAsync(_idsUri.ToString(), cancellationToken);
+            if (disco.IsError)
+                throw new ApplicationException("Failed to get discivery document");
+
+            var tokenResponse = await httpClient.RequestClientCredentialsTokenAsync(new ClientCredentialsTokenRequest
+            {
+                Address = disco.TokenEndpoint,
+
+                ClientId = "back",
+                ClientSecret = "secret",
+                Scope = "ziralink IdentityServerApi"
+            }, cancellationToken);
+
+            if (tokenResponse.IsError)
+                throw new ApplicationException("Failed to get token from identity server");
+
+            httpClient.SetBearerToken(tokenResponse.AccessToken);
+
+            return httpClient;
+        }
+
+        public async Task<T> CallIDSApis<T>(HttpClient httpClient, string relativeUri, object jsonObject, HttpMethod httpMethod, CancellationToken cancellationToken)
```

Comment:
I suggest to create a specific service for IDS with required methods

Reply:
This should not be a Achar Faranse 
![image](https://github.com/saeedmaghdam/ZiraLink.Api/assets/3509181/4b078db1-1707-42c6-819d-ad98acbc580a)

Reply:
I would rather discuss this topic with you. We can reach a good solution.